### PR TITLE
Fix parsing of invalid json

### PR DESCRIPTION
**Pull request by elnabo**
Fix trailing character in Json declaration of boolean/array/number and object see #5691

### CODE REVIEW

**Comment by elnabo** [comment]
I've added some tests, @nadako could you review this ?

**Comment by nadako** [comment]
Looks a bit too invasive to me, is there any problem with what I proposed in https://github.com/HaxeFoundation/haxe/issues/5691#issuecomment-248083020?

**Comment by elnabo** [comment]
I don't know, I had a reason at first but I'm unable to see why now, so I switched back to yours.

File: std/haxe/format/JsonParser.hx [hunk 1]
```diff
@@ -222,7 +236,7 @@ class JsonParser {
 					if (minus) minus = false;
 					digit = true; zero = false;
 				case '.'.code :
-					if (minus || point) invalidNumber(start);
+					if (minus || point || e) invalidNumber(start);
```
Comment (nadako) [inline std/haxe/format/JsonParser.hx]: what's this `e`? could you please add a test for this too?

Comment (elnabo) [reply std/haxe/format/JsonParser.hx]: According to the json spec, the exponential is an integer. So you can't have something like `1e2.5`. I'll try to add a test for that tomorrow.

**Pull request merged**